src/Vectorface/SnappyRouter/Request/HttpRequest.php

```diff
@@ -67,6 +78,17 @@ public function setVerb($verb)
     }
 
     /**
+     * Sets the stream used in the request.
+     * @param mixed $stream The stream used in the request.
+     * @return RequestInterface Returns $this.
+     */
+    public function setStream($stream)
+    {
+        $this->stream = $stream;
```

Comment:
There is stream type checking in getInputStream, but I think that type checking would be better suited here. If the arguments passed in to the setter are invalid (not a stream or path string), you can throw an InvalidArgumentException. This works nicely with the constructor too - an invalid stream passed to the constructor would abort object creation.